### PR TITLE
Hash correct file paths for global dependencies

### LINKED ISSUE #886: globalDependencies not taken into account for files in the root folder
### What version of Turborepo are you using?

1.1.6

### What package manager are you using / does the bug impact?

npm, pnpm

### What operating system are you using?

Windows

### Describe the Bug

After launching `npx create-turbo@latest prj --use-npm` then changing `turbo.json` to add `globalDependencies` for a file in the root folder, when changing the content of this file, we have a cache hit.

### Expected Behavior

We should have a cache miss.

### To Reproduce

```
npx create-turbo@latest prj --use-npm
cd prj
node -e "const c = require('./turbo.json'); c.globalDependencies= ['./file.js','../file.js','../../file.js']; require('fs').writeFileSync('./turbo.json', JSON.stringify(c, null, 2) , 'utf-8');"
echo "foo" > file.js
npm run build
echo "bar" > file.js
npm run build
```

Last `npm run build` should not use cache

### PR DESCRIPTION
**Pull request by gsoltis**
Fixes #886 

`GitHashForFiles` now takes absolute paths for all files. It was previously receiving a mix of absolute and relative paths. Also clean up execution of `git hash-objects` a little.

### CODE REVIEW

File: cli/internal/context/context.go [hunk 1]
```diff
@@ -536,11 +536,12 @@ func calculateGlobalHash(rootpath string, rootPackageJSON *fs.PackageJSON, exter
 
 	if !util.IsYarn(backend.Name) {
 		// If we are not in Yarn, add the specfile and lockfile to global deps
-		globalDeps.Add(backend.Specfile)
-		globalDeps.Add(backend.Lockfile)
+		globalDeps.Add(filepath.Join(rootpath, backend.Specfile))
+		globalDeps.Add(filepath.Join(rootpath, backend.Lockfile))
```
Comment (jaredpalmer) [inline cli/internal/context/context.go:540]: I think the absolute path will break caching across machines? Is there a different approach we can use?

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: cli/internal/context/context.go (hunk 1 above)
Comment (gsoltis) [reply cli/internal/context/context.go:540]: Good point. I suppose we can do everything relative to the repo root instead?

Comment (jaredpalmer) [reply cli/internal/context/context.go:540]: I think so. We might need to normalize paths on windows though?

Comment (gsoltis) [reply cli/internal/context/context.go:540]: Yup. We probably have this problem with `git ls-tree` too.

Comment (gsoltis) [reply cli/internal/context/context.go:540]: I made #891 a separate PR to normalize paths coming from `git ls-tree`

**Comment by jaredpalmer** [comment]
This looks great, let's fix errors in #891 and then merge these up

File: cli/internal/fs/package_deps_hash.go [hunk 2]
```diff
@@ -37,6 +38,7 @@ func GetPackageDeps(p *PackageDepsOptions) (map[string]string, error) {
 		return nil, fmt.Errorf("could not get git hashes for files in package %s: %w", p.PackagePath, err)
 	}
 	// Add all the checked in hashes.
+	// TODO(gsoltis): are these platform-dependent paths?
```
Comment (jaredpalmer) [inline cli/internal/fs/package_deps_hash.go:41]: Do we need this todo anymore?

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**